src/Storages/Distributed/DistributedAsyncInsertDirectoryQueue.cpp

```diff
@@ -186,6 +186,15 @@ void DistributedAsyncInsertDirectoryQueue::shutdownAndDropAllData()
     fs::remove_all(path);
 }
 
+void DistributedAsyncInsertDirectoryQueue::shutdown()
+{
+    /// It's incompatible with should_batch_inserts
+    /// because processFilesWithBatching may push to the queue after shutdown
```

Comment:
Actually it should not be a problem, since this queue is only in memory and it can be finished (which will break push) only in case of shutdown, so we can simply log this error and continue.

Reply:
I mean that it may trigger a logical error: https://github.com/ClickHouse/ClickHouse/blob/9ab545e28c55ecb0dc10d0f31bec51ffe7d2a732/src/Storages/Distributed/DistributedAsyncInsertDirectoryQueue.cpp#L661-L662

Reply:
Yeah, and I was suggesting to replace this `LOGICAL_ERROR` with `LOG_WARNING` or smth